app/src/main/java/com/codesoom/assignment/AppHttpHandler.java

```diff
@@ -0,0 +1,187 @@
+        String rootPath = uriPaths[1];
+        String taskId = "";
+        String contents = "";
+        if (uriPaths.length > taskIdIdx) {
+            taskId = uriPaths[taskIdIdx];
+        }
+
+        InputStream inputStream = exchange.getRequestBody();
+        String body = new BufferedReader(new InputStreamReader(inputStream))
+                .lines()
+                .collect(Collectors.joining("\n"));
+        logger.info("method : " + method + " /  uriPath : " + uriPath + " / body : " + body);
+
+        int httpStatusCode = HttpStatusCode.OK.getStatusCode();
+        try {
+            if (taskPath.equals(rootPath)) {
+                AppResponseEntity appResponseEntity = requestProcessor(method, body, taskId);
+                contents = appResponseEntity.getContents();
+                httpStatusCode = appResponseEntity.getHttpStatusCode();
+            }
+        } catch (NoSuchElementException ne) {
+            httpStatusCode = HttpStatusCode.NOT_FOUND.getStatusCode();
+        }
+
+
+        exchange.sendResponseHeaders(httpStatusCode, contents.getBytes().length);
+        OutputStream outputStream = exchange.getResponseBody();
+        outputStream.write(contents.getBytes());
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    private AppResponseEntity requestProcessor(String method, String body, String taskId) throws IOException {
+        String contents = "";
+        int httpStatusCode = HttpStatusCode.OK.getStatusCode();
+        switch (method) {
+            case "GET":
+                contents = tasksToJSONByPath(taskId);
+                httpStatusCode = HttpStatusCode.OK.getStatusCode();
+                break;
+            case "POST":
+                if (!body.isBlank()) {
+                    Task task = toTask(body);
+                    task.setId(getTaskMaxId());
+                    tasks.add(task);
+                    httpStatusCode = HttpStatusCode.CREATED.getStatusCode();
+                    contents = tasksToJSON(task);
+                }
+                break;
+            case "DELETE":
+                deleteTask(taskId);
+                httpStatusCode = HttpStatusCode.NO_CONTENT.getStatusCode();
+                break;
+            case "PUT":
+                contents = modifyTask(taskId, body);
+                httpStatusCode = HttpStatusCode.OK.getStatusCode();
+                break;
+            default:
+                break;
+        }
+        return new AppResponseEntity(httpStatusCode, contents);
+    }
+
+    private String modifyTask(String taskId, String body) throws IOException {
+
+        String contents = "";
+        if (!"".equals(taskId)) {
+            Task findTask = findTask(taskId);
+            int tasksIdx = tasks.indexOf(findTask);
+
+            Task task = toTask(body);
+            task.setId(findTask.getId());
+            tasks.set(tasksIdx, task);
+            contents = tasksToJSON(task);
+
+        }
+        return contents;
+    }
```

Comment:
다음 코드와 비교해 봅시다. 어떤 점이 다른 것 같나요?
```suggestion
    private String modifyTask(String taskId, String body) throws IOException {
        if ("".equals(taskId)) {
            return "";
        }
        Task findTask = findTask(taskId);
        int tasksIdx = tasks.indexOf(findTask);

        Task task = toTask(body);
        task.setId(findTask.getId());
        tasks.set(tasksIdx, task);
        return tasksToJSON(task);
    }
```

Reply:
if문이 짧아서 가독성이 더 좋네요 제가 한 if문은 if문이 종료할 될 때 까지 코드 확인을 다해줘야하네요

Reply:
네 간단한 내용인데 차이가 많이 나네요 ~ if문이 길었다면 정말 힘들것같아요